dsc-linux-5.4.173/patches-2022-06/00_elba_serror_fix.patch

```diff
@@ -0,0 +1,61 @@
+From 8c2594fb037dd450241eb014b993a2878ad352ce Mon Sep 17 00:00:00 2001
+From: David Clear <dac2@pensando.io>
+Date: Fri, 26 Aug 2022 13:05:38 -0700
+Subject: [PATCH] arm64: fix platform_serror handling.
+
+The previous serror handling patch incorrectly hooked bad_mode()
+whereas it should have hooked do_serror().
+
+Signed-off-by: David Clear <dac2@pensando.io>
+---
+ arch/arm64/kernel/traps.c | 16 ++++++++--------
+ 1 file changed, 8 insertions(+), 8 deletions(-)
+
+diff --git a/arch/arm64/kernel/traps.c b/arch/arm64/kernel/traps.c
+index 4b0e02cb68e0..976682bc4977 100644
+--- a/arch/arm64/kernel/traps.c
++++ b/arch/arm64/kernel/traps.c
+@@ -782,11 +782,6 @@ const char *esr_get_class_string(u32 esr)
+ 	return esr_class_str[ESR_ELx_EC(esr)];
+ }
+ 
+-int __weak platform_bad_mode(struct pt_regs *regs, unsigned int esr)
+-{
+-	return 0;
+-}
+-
+ /*
+  * bad_mode handles the impossible case in the exception vector.
+  * This is usually fatal, but give the platform a chance to handle
+@@ -794,9 +789,6 @@ int __weak platform_bad_mode(struct pt_regs *regs, unsigned int esr)
+  */
+ asmlinkage void bad_mode(struct pt_regs *regs, int reason, unsigned int esr)
+ {
+-	if (platform_bad_mode(regs, esr))
+-		return;
+-
+ 	console_verbose();
+ 
+ 	pr_crit("Bad mode in %s handler detected on CPU%d, code 0x%08x -- %s\n",
+@@ -906,10 +898,18 @@ bool arm64_is_fatal_ras_serror(struct pt_regs *regs, unsigned int esr)
+ 	}
+ }
+ 
++int __weak platform_serror(struct pt_regs *regs, unsigned int esr)
```

Comment:
traps.h has a function prototype for `platform_bad_mode` that should also change to `platform_serror`.